mypy/checker.py

```diff
@@ -609,6 +611,10 @@ def is_implicit_any(t: Type) -> bool:
 
             # Type check body in a new scope.
             with self.binder.top_frame_context():
+                # To be used for super in visit_super in checkexpr
+                # (We don't care when it's not a method)
+                if item.arguments:
+                    self.binder.push(NameExpr('__SelfType'), item.arguments[0].variable.type)
```

Comment:
Why do we use the binder for this? This feels like a hack. Could you just use a list as a stack, or does this depend on some binder features?